File: Leetcode/Daily/06-2025/19.cpp
// Problem: 2294. Partition Array Such That Maximum Difference Is K (Medium)
// Link: https://leetcode.com/problems/partition-array-such-that-maximum-difference-is-k

class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        int ans = 1, n = nums.size();
        sort(nums.begin(), nums.end());
        int curr = nums[0];

        for (int i = 1; i < n; i++) {
            if (nums[i] - curr > k)  {
                ans++;
                curr = nums[i];
            }
        }

        return ans;
    }
};

File: Leetcode/Daily/06-2025/19.py
# Problem: 2294. Partition Array Such That Maximum Difference Is K (Medium)
# Link: https://leetcode.com/problems/partition-array-such-that-maximum-difference-is-k

class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        nums.sort()
        ans, n, curr = 1, len(nums), nums[0]
        
        i = 1
        while i < n:
            if (nums[i] - curr) > k:
                ans += 1
                curr = nums[i]
            i += 1        

        return ans;                